engine/runtime-instrument-common/src/main/java/org/enso/interpreter/instrument/IdExecutionService.java

```diff
@@ -21,40 +17,69 @@
 import org.enso.logger.masking.MaskedString;
 import org.enso.pkg.QualifiedName;
 
+import com.oracle.truffle.api.CallTarget;
+import com.oracle.truffle.api.instrumentation.EventBinding;
+import com.oracle.truffle.api.instrumentation.ExecutionEventNodeFactory;
+import com.oracle.truffle.api.nodes.RootNode;
+
 public interface IdExecutionService {
   String INSTRUMENT_ID = "id-value-extractor";
 
+  public interface Callbacks {
```

Comment:
@hubertp , @4e6: one of the goals on the path towards running language server as guest Java/JS/Enso code
- #7683 

is to identify what kind of interface we need between the engine and the guest code to track execution. This is the interface. These four callback methods are enough to implement the _execution tracking functionality_ of language server.

Next step. Use them from Enso, somehow.